src/datasets/arrow_dataset.py

```diff
@@ -3436,15 +3438,22 @@ def train_test_split(
                         ),
                     }
                 )
-
         if not shuffle:
+            if stratify is not None:
+                raise ValueError("Stratified train/test split is not implemented for shuffle=False")
             train_indices = np.arange(n_train)
             test_indices = np.arange(n_train, n_train + n_test)
         else:
+            # stratified partition
+            if stratify is not None:
+                train_indices, test_indices = next(
+                    generate_indices(self[stratify], n_train, n_test, random_state=seed)
```

Comment:
This will boost performance a bit ;)
Arrow does zero-copy conversion to numpy

```suggestion
                    generate_indices(self.with_format("numpy")[stratify], n_train, n_test, random_state=seed)
```

Reply:
We instantiate the random number generator above using the seed already
```suggestion
                    generate_indices(self[stratify], n_train, n_test, random_state=generator)
```

Reply:
changed as per the suggestions.